modules/search/class.jetpack-search.php

```diff
@@ -0,0 +1,1314 @@
+							}
+						}
+
+						break;
+
+					case 'date_histogram':
+						$timestamp = $item['key'] / 1000;
+
+						$current_year  = $query->get( 'year' );
+						$current_month = $query->get( 'monthnum' );
+						$current_day   = $query->get( 'day' );
+
+						switch ( $this->aggregations[ $label ]['interval'] ) {
+							case 'year':
+								$year = (int) date( 'Y', $timestamp );
+
+								$query_vars = array(
+									'year'     => $year,
+									'monthnum' => false,
+									'day'      => false,
+								);
+
+								$name = $year;
+
+								// Is this year currently selected?
+								if ( ! empty( $current_year ) && (int) $current_year === $year ) {
+									$active = true;
+
+									$remove_url = remove_query_arg( array( 'year', 'monthnum', 'day' ) );
+								}
+
+								break;
+
+							case 'month':
+								$year  = (int) date( 'Y', $timestamp );
+								$month = (int) date( 'n', $timestamp );
+
+								$query_vars = array(
+									'year'     => $year,
+									'monthnum' => $month,
+									'day'      => false,
+								);
+
+								$name = date( 'F Y', $timestamp );
+
+								// Is this month currently selected?
+								if ( ! empty( $current_year ) && (int) $current_year === $year &&
+								     ! empty( $current_month ) && (int) $current_month === $month ) {
+									$active = true;
+
+									$remove_url = remove_query_arg( array( 'monthnum', 'day' ) );
+								}
+
+								break;
+
+							case 'day':
+								$year  = (int) date( 'Y', $timestamp );
+								$month = (int) date( 'n', $timestamp );
+								$day   = (int) date( 'j', $timestamp );
+
+								$query_vars = array(
+									'year'     => $year,
+									'monthnum' => $month,
+									'day'      => $day,
+								);
+
+								$name = date( 'F jS, Y', $timestamp );
+
+								// Is this day currently selected?
+								if ( ! empty( $current_year ) && (int) $current_year === $year &&
+								     ! empty( $current_month ) && (int) $current_month === $month &&
+								     ! empty( $current_day ) && (int) $current_day === $day ) {
+									$active = true;
+
+									$remove_url = remove_query_arg( array( 'day' ) );
+								}
+
+								break;
+
+							default:
+								continue 3; // switch() is considered a looping structure
+						} // End switch().
+
+						break;
+
+					default:
+						//continue 2; // switch() is considered a looping structure
+				} // End switch().
+
+				// Need to urlencode param values since add_query_arg doesn't
+				$url_params = urlencode_deep( $query_vars );
+
+				$aggregations_data[ $label ]['buckets'][] = array(
+					'url'        => add_query_arg( $url_params ),
+					'query_vars' => $query_vars,
+					'name'       => $name,
+					'count'      => $item['doc_count'],
+					'active'     => $active,
+					'remove_url' => $remove_url,
+					'type'       => $type,
+					'type_label' => $label,
+				);
+			} // End foreach().
+		} // End foreach().
+
+		return $aggregations_data;
```

Comment:
This var $aggregations_data might not have been defined but should be enough initializing it before the `foreach`.